components/binding/internal/webhook/binding/mutate.go

```diff
@@ -0,0 +1,65 @@
+package binding
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/kyma-project/kyma/components/binding/internal/common"
+	"github.com/kyma-project/kyma/components/binding/pkg/apis/v1alpha1"
+	"net/http"
+
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var _ admission.Handler = &MutationHandler{}
+var _ admission.DecoderInjector = &MutationHandler{}
+
+type MutationHandler struct {
+	decoder *admission.Decoder
+	log     log.FieldLogger
+}
+
+func NewMutationHandler(log log.FieldLogger) *MutationHandler {
+	return &MutationHandler{
+		log:    log,
+	}
+}
+
+func (h *MutationHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	h.log.Infof("start handling binding: %s", req.UID)
+
+	binding := &v1alpha1.Binding{}
+	if err := common.MatchKinds(binding, req.Kind); err != nil {
+		h.log.Errorf("kind does not match: %s", err)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if err := h.decoder.Decode(req, binding); err != nil {
+		h.log.Errorf("cannot decode Binding: %s", err)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if err := h.mutateBinding(ctx, binding); err != nil {
+		h.log.Errorf("cannot mutate Binding: %s", err)
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	rawBinding, err := json.Marshal(binding)
+	if err != nil {
+		h.log.Errorf("cannot marshal mutated binding: %s", err)
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	h.log.Infof("finish handling binding: %s", req.UID)
+	return admission.PatchResponseFromRaw(req.Object.Raw, rawBinding)
+}
+
+func (h *MutationHandler) InjectDecoder(d *admission.Decoder) error {
+	h.decoder = d
+	return nil
+}
+
+func (h *MutationHandler) mutateBinding(ctx context.Context, binding *v1alpha1.Binding) error {
+	binding.Finalizers = []string{v1alpha1.BindingFinalizer}
```

Comment:
I know it is our type but IMO good practice is to check if some finalizer already exist and add a new one instead override the old one